doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/ProcessDocsAction.kt

```diff
@@ -0,0 +1,276 @@
+package nl.jolanrensen.docProcessor
+
+import nl.jolanrensen.docProcessor.ProcessDocsAction.Parameters
+import nl.jolanrensen.docProcessor.gradle.ProcessDocsGradleAction
+import org.jetbrains.dokka.CoreExtensions
+import org.jetbrains.dokka.DokkaConfigurationImpl
+import org.jetbrains.dokka.DokkaGenerator
+import org.jetbrains.dokka.DokkaSourceSetImpl
+import org.jetbrains.dokka.base.DokkaBase
+import org.jetbrains.dokka.base.translators.descriptors.DefaultDescriptorToDocumentableTranslator
+import org.jetbrains.dokka.base.translators.psi.DefaultPsiToDocumentableTranslator
+import org.jetbrains.dokka.model.WithSources
+import org.jetbrains.dokka.model.withDescendants
+import org.jetbrains.dokka.utilities.DokkaConsoleLogger
+import java.io.File
+import java.util.*
+
+/**
+ * Process docs action.
+ *
+ * When [process] is executed, it will:
+ *  - analyse the sources with dokka to get the documentables ([analyseSourcesWithDokka])
+ *  - Use a [ServiceLoader] to get all available [DocProcessors][DocProcessor] ([findProcessors])
+ *  - Filter found [DocProcessors][DocProcessor] by [Parameters.processors] ([findProcessors])
+ *  - Run filtered [DocProcessors][DocProcessor] on the documentables in order of [Parameters.processors] ([process])
+ *  - Collect all modified documentables per file ([getModifiedDocumentablesPerFile])
+ *  - Write files to the [Parameters.target] directory containing modified text ([copyAndModifySources])
+ *
+ * @see [ProcessDocsGradleAction] for a Gradle specific implementation.
+ */
+abstract class ProcessDocsAction : SimpleLogger {
+
+    interface Parameters {
```

Comment:
It needs to be an interface for `ProcessDocsGradleAction`, which needs a mutable variant of this interface. I just put it inside to not clutter the namespace. I could remove the invoke factory functions, since it's an abstract class anyways.